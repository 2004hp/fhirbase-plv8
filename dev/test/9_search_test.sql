--db:fhirb
--{{{
select regexp_split_to_table from regexp_split_to_table('a,b,c', ',');
--}}}
--{{{
    SELECT parse_search_params('Patient', $JSON${
      "name": "Chalmers",
      "given": "Peter"
    }$JSON$)
--}}}

--{{{
    SELECT parse_search_params('Patient','{}'::jsonb);
--}}}
--{{{
select * from patient_search_string
--}}}

--{{{
SELECT
  search_resource('Patient',
    $JSON${
        "name": "Donald",
        "given": "Du"
    }$JSON$)

--}}}

--{{{
SELECT
  search_resource('Patient','{}');;

--}}}

--{{{
  SELECT  'SELECT ? FROM '
  || string_agg((x.row)->>'param' || '_idx', ',')
  || ' WHERE '
  FROM (
    SELECT json_array_elements($JSON$[
     {"param":"given", "value":["peter", "wolf"], "modificator": "exact"},
     {"param":"given", "value":["peter"], "modificator": "exact"},
     {"param":"telecom", "value":["work"]}
    ]$JSON$::json) as row
  ) x;
--}}}

--{{{
SELECT * from
patient_search_string s1,
patient_search_string s2
where
s1.param='given' and s1.value ilike 'peter%'
and s1.resource_id = s2.resource_id  and s2.param = 'telecom' and s2.value ilike '%home%'
--group by s1.resource_id


--}}}
