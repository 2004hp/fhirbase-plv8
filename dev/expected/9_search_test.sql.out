select regexp_split_to_table from regexp_split_to_table('a,b,c', ',');
 regexp_split_to_table 
-----------------------
 a
 b
 c
(3 rows)

SELECT parse_search_params('Patient', $JSON${
      "name": "Chalmers",
      "given": "Peter",
      "active": "true",
      "gender": "http://hl7.org/fhir/v3/AdministrativeGender|M"
    }$JSON$);
SELECT parse_search_params('Patient','{}'::jsonb);
select * from patient_search_string


SELECT
  search_resource('Patient',
    $JSON${
        "name": "Donald",
        "given": "Du"
    }$JSON$)


 SELECT '2013-01-14T10:00+0600'::timestamp;
SELECT '2013-01-14T10:00+0600'::timestamptz;
      timestamptz       
------------------------
 2013-01-14 08:00:00+04
(1 row)

SELECT '2013-01-14T10:00Z'::timestamptz;
      timestamptz       
------------------------
 2013-01-14 14:00:00+04
(1 row)

SELECT
  history_resource('Patient',
    (select logical_id from patient limit 1))


  SELECT  'SELECT ? FROM '
  || string_agg((x.row)->>'param' || '_idx', ',')
  || ' WHERE '
  FROM (
    SELECT json_array_elements($JSON$[
     {"param":"given", "value":["peter", "wolf"], "modificator": "exact"},
     {"param":"given", "value":["peter"], "modificator": "exact"},
     {"param":"telecom", "value":["work"]}
    ]$JSON$::json) as row
  ) x;
SELECT * from
patient_search_string s1,
patient_search_string s2
where
s1.param='given' and s1.value ilike 'peter%'
and s1.resource_id = s2.resource_id  and s2.param = 'telecom' and s2.value ilike '%home%'


 _id |             resource_id              | param | value | _id |             resource_id              |  param  | value 
-----+--------------------------------------+-------+-------+-----+--------------------------------------+---------+-------
  57 | 48193f40-e197-4aef-adde-3c2e85cecef8 | given | Peter |  62 | 48193f40-e197-4aef-adde-3c2e85cecef8 | telecom | home
  66 | abc30cd0-66b6-4b5e-b08e-67ae41381386 | given | Peter |  71 | abc30cd0-66b6-4b5e-b08e-67ae41381386 | telecom | home
  75 | 176e7757-55ca-4fc9-a224-ad2ef89c99dd | given | Peter |  80 | 176e7757-55ca-4fc9-a224-ad2ef89c99dd | telecom | home
  84 | 999ba67f-821c-4b8c-ae39-2ff610513d6b | given | Peter |  89 | 999ba67f-821c-4b8c-ae39-2ff610513d6b | telecom | home
(4 rows)

